#! /bin/sh /usr/share/dpatch/dpatch-run
## Patch: 21_newlist_help.patch
## Author: Tollef Fog Heen <tfheen@debian.org>
## DP: make newlist be a bit friendlier.
@DPATCH@
diff -urNad mailman-2.1.9~/bin/newlist mailman-2.1.9/bin/newlist
--- mailman-2.1.9~/bin/newlist	2007-01-20 03:44:12.000000000 +0100
+++ mailman-2.1.9/bin/newlist	2007-01-26 11:53:19.853900943 +0100
@@ -87,6 +87,9 @@
 defined in your Defaults.py file or overridden by settings in mm_cfg.py).
 
 Note that listnames are forced to lowercase.
+
+The list admin address need to be a fully-qualified address, like
+owner@example.com, not just owner.
 """
 
 import sys
@@ -94,6 +97,7 @@
 import getpass
 import getopt
 import sha
+import grp
 
 import paths
 from Mailman import mm_cfg
@@ -122,6 +126,9 @@
 
 
 def main():
+    gid = grp.getgrnam(mm_cfg.MAILMAN_GROUP)[2]
+    if os.getgid() != gid and gid not in os.getgroups():
+        os.setgid(gid)
     try:
         opts, args = getopt.getopt(sys.argv[1:], 'hql:u:e:',
                                    ['help', 'quiet', 'language=',
@@ -199,7 +206,7 @@
         except Errors.BadListNameError, s:
             usage(1, _('Illegal list name: %(s)s'))
         except Errors.EmailAddressError, s:
-            usage(1, _('Bad owner email address: %(s)s'))
+            usage(1, _('Bad owner email address: %(s)s.  Owner addresses need to be fully-qualified names, like "owner@example.com", not just "owner".'))
         except Errors.MMListAlreadyExistsError:
             usage(1, _('List already exists: %(listname)s'))
 
